packages/contract/src/pay/PayIntent.sol

```diff
@@ -0,0 +1,126 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.12;
+
+import "openzeppelin-contracts/contracts/proxy/utils/Initializable.sol";
+import "openzeppelin-contracts/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import "./TokenUtils.sol";
+import "../interfaces/IDaimoBridger.sol";
+
+/// @dev Asset amount, eg $100 USDC or 0.1 ETH
+struct TokenAmount {
+    /// @dev Zero address = native asset, eg ETH
+    IERC20 addr;
```

Comment:
should we make this `address` rather than `IERC20` since this could be a native asset?